# Производные функций нескольких переменных. Часть 2

## 1. Исследовать функцию на условный экстремум

### $$U=3-8x+6y,\,если\,x^2+y^2=36$$

Составим функцию Лагранжа:

### $$L(x,y,\lambda)=3-8x+6y +\lambda \cdot (x^2+y^2-36)$$

Найдем частные производные, приравняем их к нулю и решим систему уравнений:

### $$\begin{cases}
   L'_{x}=-8+\lambda \cdot2x=0, \\
   L'_{y}=6+\lambda \cdot2y=0, \\
   L'_{\lambda}=x^2+y^2-36=0
 \end{cases}$$

Выразим из первого уравнения $x$, из второго $y$ и подставим всё в третье:

### $$\begin{cases}
   x=\frac{8}{2\lambda}, \\
   y=-\frac{6}{2\lambda}, \\
   \Bigr(\frac{8}{2\lambda}\Bigl)^2+\Bigr(-\frac{6}{2\lambda}\Bigl)^2=36
 \end{cases}\Rightarrow 
 \begin{cases}
   x=\frac{8}{2\lambda}, \\
   y=-\frac{6}{2\lambda}, \\
   \frac{100}{4\lambda^2}=36
 \end{cases}\Rightarrow 
 \begin{cases}
   x=\frac{8}{2\lambda}, \\
   y=-\frac{6}{2\lambda}, \\
   \lambda^2=\frac{25}{36}
 \end{cases}$$

### Получаем две точки: $(\frac{24}{5},-\frac{18}{5},\frac{5}{6})$ и $(-\frac{24}{5},\frac{18}{5},-\frac{5}{6})$

Найдем вторые производные:

### $$L''_{xx}=2\lambda$$
### $$L''_{yy}=2\lambda$$
### $$L''_{\lambda \lambda}=0$$
### $$L''_{xy}=L''_{yx}=0$$
### $$L''_{x\lambda}=L''_{\lambda x}=2x$$
### $$L''_{y\lambda}=L''_{\lambda y}=2y$$ 

Составим матрицу Гёссе:

### $$\begin{pmatrix}
L''_{\lambda \lambda} & L''_{\lambda x} & L''_{\lambda y}\\ 
L''_{x \lambda} & L''_{xx} & L''_{xy}\\
L''_{y \lambda} & L''_{yx} & L''_{yy}\\
\end{pmatrix}\Rightarrow
\begin{pmatrix}
0 & 2x & 2y\\ 
2x & 2\lambda & 0\\
2y & 0 & 2\lambda\\
\end{pmatrix}$$

Далее найдем определитель в общем виде, а затем уже разберемся с каждой точкой.

### $$\Delta=\begin{vmatrix}
0 & 2x & 2y\\ 
2x & 2\lambda & 0\\
2y & 0 & 2\lambda\\
\end{vmatrix}=
\begin{vmatrix}
+ & - & +\\ 
- & + & -\\
+ & - & +\\
\end{vmatrix}=0\cdot
\begin{vmatrix}
2\lambda & 0\\
0 & 2\lambda\\
\end{vmatrix}-2x\cdot
\begin{vmatrix}
2x & 0\\
2y & 2\lambda\\
\end{vmatrix}+2y\cdot
\begin{vmatrix}
2x & 2\lambda\\
2y & 0\\
\end{vmatrix}=(-2x)\cdot 4x\lambda+2y \cdot(-4y\lambda)=-8\lambda(x^2+y^2)$$


По условию задачи $x^2+y^2=36$, тогда 
### $$\Delta=-8\lambda \cdot 36 = -288\lambda$$

Т.е. знак определителя зависит только от знака $\lambda$.

Если $\lambda=\frac{5}{6}$, то $\Delta<0$, следовательно $(\frac{24}{5},-\frac{18}{5},\frac{5}{6})$ - точка минимума.

Если $\lambda=-\frac{5}{6}$, то $\Delta>0$, следовательно $(-\frac{24}{5},\frac{18}{5},-\frac{5}{6})$ - точка максимума.

### _Ответ:_ $(\frac{24}{5},-\frac{18}{5},\frac{5}{6})$ - точка минимума, $(-\frac{24}{5},\frac{18}{5},-\frac{5}{6})$ - точка максимума.

## 2. Исследовать функцию на условный экстремум

### $$U=2x^2+12xy+32y^2+15,\,если\,x^2+16y^2=64$$

Составим функцию Лагранжа:

### $$L(x,y,\lambda)=2x^2+12xy+32y^2+15 +\lambda \cdot (x^2+16y^2-64)$$

Найдем частные производные, приравняем их к нулю и решим систему уравнений:

### $$\begin{cases}
   L'_{x}=4x+12y+\lambda \cdot2x=0, \\
   L'_{y}=12x+64y+\lambda \cdot32y=0, \\
   L'_{\lambda}=x^2+16y^2-64=0
 \end{cases}$$

Данная система уравнений имеет 4 решения (спасибо сервису https://www.kontrolnaya-rabota.ru/s/equal-many/):

### $$(-4\sqrt{2},\, -\sqrt{2}, -\frac{7}{2})$$

### $$(-4\sqrt{2},\, \sqrt{2}, -\frac{1}{2})$$

### $$(4\sqrt{2},\, -\sqrt{2}, -\frac{1}{2})$$

### $$(4\sqrt{2},\, \sqrt{2}, -\frac{7}{2})$$

Найдем вторые производные:

### $$L''_{xx}=2\lambda+4$$
### $$L''_{yy}=32\lambda+64$$
### $$L''_{\lambda \lambda}=0$$
### $$L''_{xy}=L''_{yx}=12$$
### $$L''_{x\lambda}=L''_{\lambda x}=2x$$
### $$L''_{y\lambda}=L''_{\lambda y}=32y$$ 

Составим матрицу Гёссе:

### $$\begin{pmatrix}
L''_{\lambda \lambda} & L''_{\lambda x} & L''_{\lambda y}\\ 
L''_{x \lambda} & L''_{xx} & L''_{xy}\\
L''_{y \lambda} & L''_{yx} & L''_{yy}\\
\end{pmatrix}\Rightarrow
\begin{pmatrix}
0 & 2x & 32y\\ 
2x & 2\lambda+4 & 12\\
32y & 12 & 32\lambda+64\\
\end{pmatrix}$$

Далее найдем определитель в общем виде, а затем уже разберемся с каждой точкой.

### $$\Delta=\begin{vmatrix}
0 & 2x & 32y\\ 
2x & 2\lambda+4 & 12\\
32y & 12 & 32\lambda+64\\
\end{vmatrix}=
=0\cdot
\begin{vmatrix}
2\lambda+4 & 12\\
12 & 32\lambda+64\\
\end{vmatrix}-2x\cdot
\begin{vmatrix}
2x & 12\\
32y & 32\lambda+64\\
\end{vmatrix}+32y\cdot
\begin{vmatrix}
2x & 2\lambda+4\\
32y & 12\\
\end{vmatrix}=$$


### $$=-2x\cdot (64x\lambda+128x-384y)+32y\cdot(24x-64y\lambda-128y)=$$


### $$=-128\cdot(x^2\lambda+2x^2-6xy)+256\cdot(3xy-8y^2\lambda-16y^2)=-128\cdot(x^2\lambda-6xy)+256\cdot(3xy-8y^2\lambda)-256x^2-256\cdot16y^2=-128\cdot(x^2\lambda-6xy)+256\cdot(3xy-8y^2\lambda)-256\cdot(x^2+16y^2)$$


По условию задачи $x^2+16y^2=64$, тогда 
### $$\Delta=-128\cdot(x^2\lambda-6xy)+256\cdot(3xy-8y^2\lambda)-256\cdot 64$$


Найдем значения определителя для точек:

In [1]:
x=-4*(2**(1/2))
y=-(2**(1/2))
l=-7/2
-128*(l*x**2-6*x*y)+256*(3*x*y-8*l*y**2)-256*64

24576.000000000007

### $$\Delta|_{(-4\sqrt{2},\, -\sqrt{2}, -\frac{7}{2})}=24576>0\Rightarrow\,точка\,(-4\sqrt{2},\, -\sqrt{2}, -\frac{7}{2})\,-\, это\,максимум$$


In [2]:
x=-4*(2**(1/2))
y=(2**(1/2))
l=-1/2
-128*(l*x**2-6*x*y)+256*(3*x*y-8*l*y**2)-256*64

-24576.0

### $$\Delta|_{(-4\sqrt{2},\, \sqrt{2}, -\frac{1}{2})}=-24576<0\Rightarrow\,точка\,(-4\sqrt{2},\, \sqrt{2}, -\frac{1}{2})\,-\, это\,минимум$$


In [3]:
x=4*(2**(1/2))
y=-(2**(1/2))
l=-1/2
-128*(l*x**2-6*x*y)+256*(3*x*y-8*l*y**2)-256*64

-24576.0

### $$\Delta|_{(4\sqrt{2},\, -\sqrt{2}, -\frac{1}{2})}=-24576<0\Rightarrow\,точка\,(4\sqrt{2},\, -\sqrt{2}, -\frac{1}{2})\,-\, это\,минимум$$


In [4]:
x=4*(2**(1/2))
y=(2**(1/2))
l=-7/2
-128*(l*x**2-6*x*y)+256*(3*x*y-8*l*y**2)-256*64

24576.000000000007

### $$\Delta|_{(4\sqrt{2},\, \sqrt{2}, -\frac{7}{2})}=24576>0\Rightarrow\,точка\,(4\sqrt{2},\, \sqrt{2}, -\frac{7}{2})\,-\, это\,максимум$$


### _Ответ:_ $(-4\sqrt{2},\, \sqrt{2}, -\frac{1}{2}),\,(4\sqrt{2},\, -\sqrt{2}, -\frac{1}{2})$ - точки минимума, $(-4\sqrt{2},\, -\sqrt{2}, -\frac{7}{2}),\,(4\sqrt{2},\, \sqrt{2}, -\frac{7}{2})$ - точки максимума.

## 3. Найти производную функции $U=x^2+y^2+z^2$ по направлению вектора $\overrightarrow{c}(-9,\,  8,\,-12)$ в точке $M(8,\, -12,\, 9)$

Найдем длину вектора $\overrightarrow{c}$:

### $$|\overrightarrow{c}|=\sqrt{x^2_{0}+y^2_{0}+z^2_{0}}=\sqrt{(-9)^2+8^2+(-12)^2}=\sqrt{289}=17$$

Запишем единичный вектор $\overrightarrow{c}_{0}$:

### $$\overrightarrow{c_{0}}=\frac{\overrightarrow{c}}{|\overrightarrow{c}|}=\Bigr(-\frac{9}{17},\,\frac{8}{17},\,-\frac{12}{17}\Bigl)$$

Теперь найдем градиент в точке $M(8,\, -12,\, 9)$:

### $$U'_{x}=2x$$

### $$U'_{y}=2y$$

### $$U'_{z}=2z$$

### $$\mbox{grad}U=(16,\, -24,\, 18)$$

Посчитаем значение производной:

### $$U'_{\overrightarrow{с}}=-\frac{9}{17}\cdot 16+\frac{8}{17}\cdot (-24)-\frac{12}{17}\cdot 18=\frac{-144-192-216}{17}=-\frac{552}{17}$$

### _Ответ:_ $-\frac{552}{17}$

## 4. Найти производную функции $U=e^{x^2+y^2+z^2}$ по направлению вектора $\overrightarrow{d}(4,\, -13,\, -16)$ в точке $M(-16,\, 4,\, 13)$

Найдем длину вектора $\overrightarrow{d}$:

### $$|\overrightarrow{d}|=\sqrt{x^2_{0}+y^2_{0}+z^2_{0}}=\sqrt{4^2+(-13)^2+(-16)^2}=\sqrt{441}=21$$

Запишем единичный вектор $\overrightarrow{d}_{0}$:

### $$\overrightarrow{d_{0}}=\frac{\overrightarrow{d}}{|\overrightarrow{d}|}=\Bigr(\frac{4}{21},\,-\frac{13}{21},\,-\frac{16}{21}\Bigl)$$

Теперь найдем градиент в точке $M(-16,\, 4,\, 13)$:

### $$U'_{x}=e^{x^2+y^2+z^2}\cdot2x$$

### $$U'_{y}=e^{x^2+y^2+z^2}\cdot2y$$

### $$U'_{z}=e^{x^2+y^2+z^2}\cdot2z$$

### $$\mbox{grad}U=(-32e^{441},\, 8e^{441},\, 26e^{441})$$

Посчитаем значение производной:

### $$U'_{\overrightarrow{d}}=\frac{4}{21}\cdot (-32e^{441})-\frac{13}{21}\cdot 8e^{441}-\frac{16}{21}\cdot 26e^{441}=\frac{(-128-104-416)e^{441}}{21}=-\frac{648 e^{441}}{21}$$

### _Ответ:_ $-\frac{648 e^{441}}{21}$